sway-core/src/semantic_analysis/ast_node/declaration/mod.rs

```diff
@@ -1,4 +1,6 @@
 mod abi;
+#[allow(clippy::module_inception)]
+mod declaration;
```

Comment:
In the source on my machine this is still in `ast_node/declaration.rs` and it seems like the new `declaration.rs` code taken from `ast_node/mod.rs` could go in there or here in this `mod.rs` since it's so small and the module inception could be avoided.